File: gen_scripts/ruby/j12.arguments.rb
#!/usr/bin/env ruby
# illustrative variables
first     = 0  # set index of first element
min_count = 0  # set minimum number of elements required
# utility variables
count     = 0  # initialize counter

puts "The arguments passed are:"
# conditional loop to process elements using shift
while ARGV.length > min_count do          # fetches updated num of args
 puts " item #{count+=1}: #{ARGV[first]}" # output result of first argument
 ARGV.shift                               # remove first element
end

File: perl/j12.arguments.pl
#!/usr/bin/perl
# illustrative variables
my $last  = $#ARGV; # get index of last element
my $first = 0;      # set index of first element 

print "The arguments passed are:\n";
# iterative loop to enumerate args
#   Notes:
#    - $count     is index into array
#    - $count + 1 is count that is displayed to user
for ( my $count = $first; $count <= $last ; $count++ ) {
  my $arg = $ARGV[$count];  # get arg using index
  # output count and argument using count index
  printf "  item %d: %s\n", $count+1, $arg;
}